packages/@aws-cdk/aws-lambda/lib/lambda-role.ts

```diff
@@ -0,0 +1,127 @@
+import { IManagedPolicy, ManagedPolicy, Role, RoleProps, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { Construct} from '@aws-cdk/core';
+
+/**
+ * Properties for defining a Lambda Role
+ */
+export interface LambdaRoleProps extends Omit<RoleProps, 'assumedBy'> {
```

Comment:
Unfortunately, jsii doesn't support the `Omit` type.

The feature set supported in the CDK on public-facing interfaces is restricted by the feature set supported by JSII.
My guess is the Java equivalent interfaces have simply ignored the Omit and in fact, do require the 'assumedBy' property.

I've opened a bug report for this - https://github.com/aws/jsii/issues/1707

Reply:
If you want to achieve this, you will need to refactor [RoleProps](https://github.com/aws/aws-cdk/blob/e257dc823be3b7d1a2acaf0f19b88a40894a7b6a/packages/%40aws-cdk/aws-iam/lib/role.ts#L16) to create a new `interface` that holds everything in `RoleProps` except `assumedBy` and `RoleProps` would simply extend this.

```ts
export interface RoleProps extends RoleOptions {
  readonly assumedBy: IPrincipal;
}

export interface RoleOptions {
  // the rest of the properties here
}
```

I've used `RoleOptions` but we might have to use a better name.

Reply:
@nija-at will Omit ever be supported by jsii?

I'm fine with doing the refactor, it's just tedious.

Reply:
There's no reasonable way to model this in languages outside of typescript (think: Java, C#, Python, etc.), so we don't have any plans to support this.

Reply:
Fixed in recent commit